### PR TITLE
Don't invoke log_event from state machine

### PR DESCRIPTION
**Pull request by crusaderky**
- Follow-up to #6404
- Blocks #6476
- Don't catch a too-generic ValueError in `Worker._transition`
- Don't call Worker.log_event from inside the state machine. Now all calls are in the outermost layer of `Worker.handle_stimulus` and `Worker.validate_state`; they will remain in Worker post refactor, in methods that override those in `BaseWorker`.

### CODE REVIEW

File: distributed/tests/test_worker.py [hunk 1]
```diff
@@ -3332,29 +3331,6 @@ async def test_tick_interval(c, s, a, b):
...
-    while not s.events["invalid-worker-transition"]:
-        await asyncio.sleep(0.01)
-
-    assert "foo" in str(s.events["invalid-worker-transition"])
-    assert a.address in str(s.events["invalid-worker-transition"])
-    assert ts.key in str(s.events["invalid-worker-transition"])
-
-    del s.events["invalid-worker-transition"]  # for test cleanup
-
-
```
Comment (crusaderky) [inline distributed/tests/test_worker.py:3357]: Duplicates test in test_utils_test.py

**Comment by crusaderky** [comment]
Ready for final review and merge

File: distributed/scheduler.py [hunk 2]
```diff
@@ -6224,7 +6224,11 @@ async def feed(
             if teardown:
                 teardown(self, state)
 
-    def log_worker_event(self, worker=None, topic=None, msg=None):
+    def log_worker_event(
+        self, worker: str, topic: str | Collection[str], msg: Any
+    ) -> None:
+        if isinstance(msg, dict):
+            msg["worker"] = worker
```
Comment (mrocklin) [inline distributed/scheduler.py:6231]: I'm curious, why this change?

For context, this tripped me up when doing some Coiled work.  I'm not against it, I'm just curious about the motivation.

Comment (crusaderky) [reply distributed/scheduler.py:6231]: Previously, the same info was manually added to *most* worker-side calls to log-event. Adding it when it lands looks cleaner and less redundant.

Comment (mrocklin) [reply distributed/scheduler.py:6231]: Users can use this system too though.  We're adding this to their messages.

Comment (mrocklin) [reply distributed/scheduler.py:6231]: Also, when do you sleep?  I'm about to finish up my work day in California.  If you're working past when I do then that really puts me to shame :)

Comment (mrocklin) [reply distributed/scheduler.py:6231]: (travelling away from home in Texas this week)
